server.go

```diff
@@ -126,7 +150,17 @@ func Listen(conn net.PacketConn, tlsConf *tls.Config, config *Config) (Listener,
 	return listen(conn, tlsConf, config)
 }
 
-func listen(conn net.PacketConn, tlsConf *tls.Config, config *Config) (*server, error) {
+// ListenEarly works like Listen, but it returns sessions before the handshake completes.
+func ListenEarly(conn net.PacketConn, tlsConf *tls.Config, config *Config) (EarlyListener, error) {
+	s, err := listen(conn, tlsConf, config)
+	if err != nil {
+		return nil, err
+	}
+	s.acceptEarlySessions = true
```

Comment:
Is there a race condition here if a client connects before this is set?

Reply:
Right, this should better be a parameter to `listen`.